app/models/vim_performance_state/purging.rb

```diff
@@ -11,8 +11,13 @@ def purge_mode_and_value
       # remove anything where the resource no longer exists AND
       # remove anything older than a certain date
       def purge_timer
-        purge_queue(:orphaned, "resource")
-        purge_queue(:date, purge_date)
+        purge_queue(:date, purge_date, concurrent: false)
+
+        # States are purged daily via vim_performance_states_purge_interval.
+        # We purge by_date first since the date index should be faster than via resource.
+        # To avoid contention on the same table, we enqueue purge_date and purge_orphaned
+        # with a task_id so they won't be run at the same time.
+        purge_queue(:orphaned, "resource", concurrent: false)
```

Comment:
I actually feel more comfortable stating a new type `date_and_orphans`

and the method takes a date, and resource.
then it calls the 2 in order.

Reply:
I would assume we did it through separate queue items so they have their own timeouts.